src/spac/visualization.py

```diff
@@ -9,6 +9,104 @@
 from matplotlib.colors import ListedColormap, BoundaryNorm
 
 
+def boxplot(adata, annotation=None, layer=None, ax=None,
+            features=None, **kwargs):
+    """
+    Plot boxplots for all features available in the passed adata object or
+    a subset if features are provided.
+
+    Parameters
+    ----------
+    adata : anndata.AnnData
+        The AnnData object.
+
+    annotation : str, optional
+        Annotation to determine if separate plots are needed for every label.
+
+    layer : str, optional
+        The name of the matrix layer to use. If not provided,
+        uses the main data matrix adata.X.
+
+    ax : matplotlib.axes.Axes, optional
+        An existing Axes object to draw the plot onto, optional.
+
+    features : list, optional
+        List of feature names to be plotted.
+        If not provided, all features will be plotted.
+
+    **kwargs
+        Additional arguments to pass to seaborn.boxplot.
+
+    Returns
+    -------
+    fig, ax : matplotlib.figure.Figure, matplotlib.axes.Axes
+        The created figure and axes for the plot.
+    """
+
+    # Validate inputs
+    if not isinstance(adata, anndata.AnnData):
+        raise TypeError("Input 'adata' must be an instance "
+              "of anndata.AnnData.")
+
+    if ax and not isinstance(ax, plt.Axes):
+        raise TypeError("Input 'ax' must be a matplotlib.axes.Axes object.")
+
+    if annotation and annotation not in adata.obs.columns:

```

Comment:
Fang, please use the standard utility checks that Rui implemented: https://github.com/FNLCR-DMAP/SCSAWorkflow/blob/dev/src/spac/utils.py#L202